File: src/main/java/net/igneo/icv/networking/packet/EnchantUseC2SPacket.java
package net.igneo.icv.networking.packet;

import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.server.level.ServerLevel;
import net.minecraft.server.level.ServerPlayer;
import net.minecraftforge.network.NetworkEvent;

import java.util.function.Supplier;

import static net.igneo.icv.event.ModEvents.useEnchant;


public class EnchantUseC2SPacket {
    private int slot;
    public EnchantUseC2SPacket(int slot){
        this.slot = slot;
    }
    public EnchantUseC2SPacket(FriendlyByteBuf buf) {
        this.slot = buf.readInt();
    }

    public void toBytes(FriendlyByteBuf buf) {
        buf.writeInt(this.slot);
    }

    public boolean handle(Supplier<NetworkEvent.Context> supplier) {
        NetworkEvent.Context context = supplier.get();
        context.enqueueWork(() -> {
            ServerLevel level = context.getSender().serverLevel();
            ServerPlayer player = context.getSender();
            useEnchant(player, slot);
        });
        return true;
    }
}

File: src/main/java/net/igneo/icv/networking/packet/EnchantUseC2SPacket.kt
package net.igneo.icv.networking.packet

import net.igneo.icv.Utils.useEnchant
import net.minecraft.network.FriendlyByteBuf
import net.minecraftforge.network.NetworkEvent
import java.util.function.Supplier

class EnchantUseC2SPacket {
    private val slot: Int

    constructor(slot: Int) {
        this.slot = slot
    }

    constructor(buf: FriendlyByteBuf) {
        this.slot = buf.readInt()
    }

    fun toBytes(buf: FriendlyByteBuf) {
        buf.writeInt(this.slot)
    }

    fun handle(supplier: Supplier<NetworkEvent.Context>): Boolean {
        val context = supplier.get()
        context.enqueueWork {
            context.sender!!.serverLevel()
            val player = context.sender
            println("using on the server")
            useEnchant(player!!, slot)
        }
        return true
    }
}
